src/Persistent/Pure.hs

```diff
@@ -709,8 +710,20 @@ scopeHiearchy = \case
             IdeaLocationSpace s    -> DScopeIdeaSpace s
             IdeaLocationTopic _s t -> DScopeTopicId   t)
 
-findDelegationsByScope :: DScope -> Query [(Delegate (AUID User), DScope, [Delegatee (AUID User)])]
-findDelegationsByScope = views dbDelegations . Data.Delegation.findDelegationsByScope
+findDelegationsByScope :: DScope -> Query [(Delegate (AUID User), [(DScope, Delegatee (AUID User))])]
```

Comment:
it seems to me you are nesting the list here just to flatten it out again later above.  i think i'll revert that.